### PR TITLE
Use the `layout` as specified in CSS 

### PR DESCRIPTION
**Pull request by darrenburns**
We no longer create a `DockLayout()` in the code, and instead use the value specified in CSS.

I've added a new rule to the View-level CSS to add `layout: dock` so that it's used by default without you having to specify it in the CSS (and so `basic.css` works without changes).

### CODE REVIEW

File: src/textual/app.py [hunk 1]
```diff
@@ -510,7 +503,7 @@ async def on_event(self, event: events.Event) -> None:
         # Handle input events that haven't been forwarded
         # If the event has been forwarded it may have bubbled up back to the App
         if isinstance(event, events.Mount):
-            view = View()
+            view = View(id="_root")
```
Comment (willmcgugan) [inline src/textual/app.py:506]: Why the id of _root ?

File: src/textual/css/_style_properties.py [hunk 2]
```diff
@@ -290,6 +290,39 @@ def __set__(self, obj: Styles, spacing: str | None) -> str | None:
...
+    def __set__(self, obj: Styles, layout: LayoutName | Layout):
+        """
+        Args:
+            obj (Styles): The Styles object.
+            layout (LayoutName | Layout): The layout to use. You can supply a ``LayoutName``
+                (a string literal such as ``"dock"``) or a ``Layout`` object.
+        """
+        from ..layouts.factory import get_layout, Layout  # Prevents circular import
+
+        obj.refresh(True)
```
Comment (willmcgugan) [inline src/textual/css/_style_properties.py:318]: Can you set make keyword args explicity, i.e. `layout=True`

File: src/textual/css/_styles_builder.py [hunk 3]
```diff
@@ -288,7 +289,16 @@ def process_layout(self, name: str, tokens: list[Token], important: bool) -> Non
...
-                self.styles._rule_layout = tokens[0].value
+                value = tokens[0].value
+                layout_name = cast(LayoutName, value)
+                try:
+                    self.styles._rule_layout = get_layout(layout_name)
+                except MissingLayout:
+                    self.error(
+                        name,
+                        tokens[0],
+                        f"invalid value for layout (received {value!r}, expected {friendly_list(LAYOUT_MAP.keys())})",
```
Comment (willmcgugan) [inline src/textual/css/_styles_builder.py:300]: Nice. Very much in keeping with my helpful error messages policy.

File: src/textual/css/stylesheet.py [hunk 4]
```diff
@@ -108,25 +108,44 @@ def _check_rule(cls, rule: RuleSet, node: DOMNode) -> Iterable[Specificity3]:
                 yield selector_set.specificity
 
     def apply(self, node: DOMNode) -> None:
+        """
```
Comment (willmcgugan) [inline src/textual/css/stylesheet.py:111]: Can you add a brief description in docstrings. Even if it seems a little redundant.

**Review by willmcgugan — APPROVED** [review]
Looks good. Just a few requests, and a question.

File: src/textual/app.py (hunk 1 above)
Comment (darrenburns) [reply src/textual/app.py:506]: Think this is just a leftover from debugging. I was printing out the node IDs from the code that walks the DOM tree and was wondering why one of them was printing `None`. Turns out it was the this view, which looks like the root of the tree.

File: src/textual/css/_style_properties.py (hunk 2 above)
Comment (darrenburns) [reply src/textual/css/_style_properties.py:318]: Done (also fixed all other occurrences in this file 😄)

**Pull request merged**